lib/Doctrine/DBAL/Driver/PDOSqlsrv/Connection.php

```diff
@@ -33,8 +35,10 @@ public function lastInsertId($name = null)
 
         $stmt = $this->prepare('SELECT CONVERT(VARCHAR(MAX), current_value) FROM sys.sequences WHERE name = ?');
         $stmt->execute([$name]);
+        $lastInsertId = $stmt->fetchColumn();
+        assert(is_string($lastInsertId));
```

Comment:
This doesn't look valid. If the sequence doesn't exist, the query will return `false`.

Reply:
Not sure why I did this, reverting it doesn't make Psalm fail :thinking:

Reply:
Ah wait that was Phpstan:

```
 ------ ------------------------------------------------------------------ 
  Line   Doctrine/DBAL/Driver/PDOSqlsrv/Connection.php                     
 ------ ------------------------------------------------------------------ 
  37     Method Doctrine\DBAL\Driver\PDOSqlsrv\Connection::lastInsertId()  
         should return string but returns string|false|null.               
 ------ ------------------------------------------------------------------ 
```

Reply:
Ok I will assert that it's either a string or `false`

Reply:
It's better to assert that it's [not null](https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-sequences-transact-sql?view=sql-server-ver15) (see the column type).

Reply:
This might be an LSP violation actually: this method should only ever return strings according to [the docs](https://www.php.net/manual/en/pdo.lastinsertid.php) (and also the phpstorm stubs)
Should we ignore this particular violation? Or should we return `'0'` if not a string?

Reply:
The documentation on PHP is really vague because it doesn't describe any non-happy paths except for "not implemented". Since it's `2.x`, let's suppress the error instead of modifying the behavior in any way.